pkg/controller/knativeserving/common/extensions.go

```diff
@@ -67,6 +73,26 @@ func (exts Extensions) Transform(instance *servingv1alpha1.KnativeServing) []mf.
 	})
 }
 
+func updateDeployment(scheme *runtime.Scheme, instance *servingv1alpha1.KnativeServing, u *unstructured.Unstructured) error {
+
+	var deployment = &appsv1.Deployment{}
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, deployment)
```

Comment:
is this better than `scheme.Convert(u, deployment)`?

Reply:
I couldn't get `scheme.Convert(u, deployment)` to work. I got error message `TypeMeta not present in src` which comes from https://github.com/kubernetes/apimachinery/blob/961b39a1baa06f6c52bdd048a809b9f5b47f1337/pkg/conversion/converter.go#L828. The issue I believe is `TypeMeta` https://github.com/knative/serving-operator/blob/6c0272eca46470454567e78aafbfd53cf7a8f53c/vendor/k8s.io/api/apps/v1/types.go#L255 is not in the unstructured map. It's unable to take the inline json content and convert it to a Deployment. This way worked. 

I'll look more into the scheme tests and see if I can reproduce this issue: https://github.com/kubernetes/apimachinery/blob/961b39a1baa06f6c52bdd048a809b9f5b47f1337/pkg/runtime/scheme_test.go#L1